src/simplifier.cpp

```diff
@@ -0,0 +1,57 @@
+#include "simplifier.h"
+#include "ir/visitor.h"
+#include "pruner_util.h"
+
+namespace P4PRUNER {
+
+IR::Node *Simplifier::preorder(IR::Operation_Binary *s) {
```

Comment:
Let's also include `Operation_Unary` and `Operation_Ternary` in this. Otherwise not all expressions will be visited. 

There is also `MethodCallExpression` and `StructExpression` we should visit. Essentially we just need to visit the children there.

Reply:
I couldn't figure out how to visit the children of a `MethodCallExpression`